From 145c1606e1658e116174ac05325b16a8e0b75291 Mon Sep 17 00:00:00 2001
From: Madhusudhan S <madhusudhan.s@intel.com>
Date: Mon, 12 Nov 2018 13:18:34 +0530
Subject: [PATCH] Enabling suspend on IVI after clicking sleep from the power
 button menu.

Tracked-on: OAM-56502

Change-Id: Ib8e5f351815474d8e99739938ac5845227f711ff
Signed-off-by: Madhusudhan S <madhusudhan.s@intel.com>

Conflicts:
	services/core/java/com/android/server/power/PowerManagerService.java

Conflicts:
	services/core/java/com/android/server/power/PowerManagerService.java

Conflicts:
	services/core/java/com/android/server/power/PowerManagerService.java
---
 .../java/com/android/server/power/PowerManagerService.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/services/core/java/com/android/server/power/PowerManagerService.java b/services/core/java/com/android/server/power/PowerManagerService.java
index 3d65ede9500..50cdc0dc4c8 100644
--- a/services/core/java/com/android/server/power/PowerManagerService.java
+++ b/services/core/java/com/android/server/power/PowerManagerService.java
@@ -119,6 +119,7 @@ import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Objects;
 import java.util.Set;
+import android.os.FileUtils;
 
 /**
  * The power manager service is responsible for coordinating power management
@@ -1775,6 +1776,12 @@ public final class PowerManagerService extends SystemService
                     Math.max(reason, PowerManager.GO_TO_SLEEP_REASON_MIN));
             Slog.i(TAG, "Going to sleep due to " + PowerManager.sleepReasonToString(reason)
                     + " (uid " + uid + ")...");
+			// Adding force suspend code to enter S3 after pressing sleep button
+			try {
+				FileUtils.stringToFile("/sys/power/state", "mem");
+			} catch (IOException e) {
+				Slog.v(TAG, "IOException: " + e);
+			}
 
             mLastSleepTime = eventTime;
             mLastSleepReason = reason;
-- 
2.17.1

